Applications/Spire/Include/Spire/Ui/DropDownMenu2.hpp

```diff
@@ -0,0 +1,132 @@
+#ifndef SPIRE_DROP_DOWN_MENU_2_HPP
+#define SPIRE_DROP_DOWN_MENU_2_HPP
+#include <Beam/SignalHandling/ConnectionGroup.hpp>
+#include <QVBoxLayout>
+#include "Spire/Spire/Spire.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/DropDownItem.hpp"
+#include "Spire/Ui/DropDownWindow.hpp"
+#include "Spire/Ui/ScrollArea.hpp"
+
+namespace Spire {
+
+  //! Represents a Spire-styled drop down menu.
+  class DropDownMenu2 : public DropDownWindow {
+    public:
+
+      //! Signals that the menu was closed.
+      using ClosedSignal = Signal<void ()>;
```

Comment:
What is the use case for this signal?

Reply:
If a listener is using the hovered and current signals to control a preview feature, then the closed signal is used as a trigger to revert whatever the preview value is to the currently selected value.

Reply:
The trigger should be when the argument to the hovered/current signal is ```!QVariant::is_valid()```